### PR TITLE
feat: allow to edit tags in settings

### PR DESCRIPTION
**Pull request by khalilJayab**
Hi,
This pull request addresses issue #6172 , which was related to the inability to rename tags in the settings. With this fix, users can now easily edit the name of existing tags in the settings section.

Untitled

Thanks

PART OF GTC OPEN SOURCE INTIATIVE

### CODE REVIEW

File: app/Http/Controllers/SettingsController.php [hunk 1]
```diff
@@ -363,6 +365,36 @@ public function deleteTag($tagId)
...
+    /**
+     * Method editTag
+     *
+     * @param Tag $tag 
+     * @param Request $request
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    
+    public function editTag(Tag $tag, Request $request)
```
Comment (kniziol) [inline app/Http/Controllers/SettingsController.php:378]:
```suggestion
    public function editTag(Tag $tag, Request $request): RedirectResponse
```
Shall we add the return type?

File: resources/views/settings/tags.blade.php [hunk 2]
```diff
@@ -72,6 +76,10 @@
                     </ul>
                   </div>
                   <div class="table-cell actions">
+                    <a href="javascript:;" class="edit-icon-btn" onclick="toggleEditInput(this)">
```
Comment (kniziol) [inline resources/views/settings/tags.blade.php:79]:
```suggestion
                    <div class="edit-icon-btn" onclick="toggleEditInput(this)">
```
Will it work? In this way we can avoid `href="javascript:;"`.

File: resources/views/settings/tags.blade.php [hunk 3]
```diff
@@ -94,3 +102,15 @@
 </div>
 
 @endsection
+<script>
+  function toggleEditInput(editIcon) {
+    var tableRow = editIcon.closest('.table-row');
```
Comment (kniziol) [inline resources/views/settings/tags.blade.php:107]:
```suggestion
    let tableRow = editIcon.closest('.table-row');
```
The `let` is recommended when declaring variables
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

**Review by kniziol — COMMENTED** [review]
Added some suggestions

**Comment by khalilJayab** [comment]
Done, thanks

File: resources/views/settings/tags.blade.php [hunk 4]
```diff
@@ -94,3 +102,15 @@
 </div>
 
 @endsection
+<script>
+  function toggleEditInput(editIcon) {
+    let tableRow = editIcon.closest('.table-row');
+    var tagName = tableRow.querySelector('.tag-name');
```
Comment (kniziol) [inline resources/views/settings/tags.blade.php:108]: Would mind you replacing all the other `var`s with `let`s as well? Thanks.

Comment (khalilJayab) [reply resources/views/settings/tags.blade.php]: done

File: app/Http/Controllers/SettingsController.php [hunk 5]
```diff
@@ -363,6 +365,36 @@ public function deleteTag($tagId)
...
+    
+    /**
+     * Method editTag
+     *
+     * @param Tag $tag 
+     * @param Request $request
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+
```
Comment (kniziol) [inline app/Http/Controllers/SettingsController.php:377]:
```suggestion
```
Just one, minor thing

**Review by asbiin — APPROVED** [review]

**Pull request merged**